logstash-core/src/main/java/org/logstash/execution/inputs/HttpPoller.java

```diff
@@ -0,0 +1,73 @@
+package org.logstash.execution.inputs;
+
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Map;
+import org.logstash.execution.Input;
+import org.logstash.execution.LsConfiguration;
+import org.logstash.execution.LsContext;
+import org.logstash.execution.PluginConfigSpec;
+import org.logstash.execution.QueueWriter;
+
+public final class HttpPoller implements Input {
+
+    private static final PluginConfigSpec<String> USER_CONFIG =
+        LsConfiguration.stringSetting("user");
+
+    private static final PluginConfigSpec<String> PASSWORD_CONFIG =
+        LsConfiguration.stringSetting("password");
+
+    private static final PluginConfigSpec<Long> AUTOMATIC_RETRIES_CONFIG =
+        LsConfiguration.numSetting("automatic_retries", 1L);
+
+    private static final PluginConfigSpec<Path> CA_CERT_CONFIG =
+        LsConfiguration.pathSetting("cacert");
+
+    private static final PluginConfigSpec<String> URL_METHOD_CONFIG =
+        LsConfiguration.stringSetting("method");
+
+    private static final PluginConfigSpec<Map<String, LsConfiguration>> URLS_CONFIG =
+        LsConfiguration.requiredHashSetting(
+            "urls", Arrays.asList(URL_METHOD_CONFIG, USER_CONFIG)
```

Comment:
I don't see how this encapsulates an HTTP poller configuration. `urls` can either look like:

`urls = {"elastic" => "http://elastic.co"}`

or

`urls = {"elastic" => {"method" => "get", "url" => "http://elastic.co", "headers" => {...}}}`

Reply:
@andrewvc hah you're right :) The "or" case that allows for two different schemas isn't covered by my proposal.

Tbh. ... I'm tempted to vote for not allowing that in Java plugins. The only downside to making that simplification is, that we'd have to adjust the config schema for the HTTP poller if we fully port it to Java (and the HTTP poller probably isn't the only case). Other than that I only see upsides to simplifying this (especially when thinking about the UI!) ...

Think it's feasible to break with the flexibility of the Ruby implementation here for the sake of type safety?

Reply:
Wow, I never responded here. At any rate, yes, I think it's fine to break with Ruby here.